source/extensions/tracers/xray/config.cc

```diff
@@ -0,0 +1,48 @@
+#include "extensions/tracers/xray/config.h"
+
+#include <string>
+
+#include "envoy/registry/registry.h"
+
+#include "common/common/utility.h"
+#include "common/config/datasource.h"
+#include "common/tracing/http_tracer_impl.h"
+
+#include "extensions/tracers/well_known_names.h"
+#include "extensions/tracers/xray/xray_tracer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+XRayTracerFactory::XRayTracerFactory() : FactoryBase(TracerNames::get().XRay) {}
+
+Tracing::HttpTracerPtr
+XRayTracerFactory::createHttpTracerTyped(const envoy::config::trace::v2::XRayConfig& proto_config,
+                                         Server::Instance& server) {
+  std::string sampling_rules_json;
+  try {
+    sampling_rules_json =
+        Config::DataSource::read(proto_config.sampling_rule_manifest(), true, server.api());
+  } catch (EnvoyException& e) {
+    ENVOY_LOG(error, "Could not read custom sampling rule json file: {}, because of {}.",
+              proto_config.sampling_rule_manifest().filename(), e.what());
```

Comment:
I don't think you can log "filename()" anymore since it may not be set depending on the data source type.

Reply:
good catch - updated.